nucypher/crypto/powers.py

```diff
@@ -180,6 +184,12 @@ def sign_transaction(self, unsigned_transaction: dict) -> HexBytes:
         signed_raw_transaction = self.blockchain.client.sign_transaction(transaction=unsigned_transaction)
         return signed_raw_transaction
 
+    def __enter__(self):
+        return self.unlock_account()
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
```

Comment:
Nice!
Though keep in mind that unlocking is a relatively slow process: kdf performed by geth eats a lot of CPU. So, I think, the default behavior (keep unlocked for 5 min) is actually good for often transactions

Reply:
Perhaps, a sensible thing is to use this context manager for confirming activity, but to rely on automatic locking for other transactions (e.g. challenges)